lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4323,25 +4325,36 @@ namespace {
       // Check whether there is a function with the same name as this
       // property. If so, suppress the property; the user will have to use
       // the methods directly, to avoid ambiguities.
-      auto containerTy = dc->getDeclaredTypeInContext();
+      Type containerTy;
+      if (auto *classDecl = dyn_cast<ClassDecl>(dc)) {
+        // If we're importing into the primary @interface for something, as
+        // opposed to an extension, make sure we don't try to load any
+        // categories...by just looking into the super type.
+        containerTy = classDecl->getSuperclass();
```

Comment:
Hm. I don't actually know about protocols, but we don't *currently* look at them. It probably helps that we mirror all their members down for imported Clang things.